api/src/paths/permit-no-sampling.ts

```diff
@@ -0,0 +1,125 @@
+import { RequestHandler } from 'express';
+import { Operation } from 'express-openapi';
+import { WRITE_ROLES } from '../constants/roles';
+import { getDBConnection, IDBConnection } from '../database/db';
+import { IPostPermitNoSampling, PostPermitNoSamplingObject } from '../models/permit-no-sampling';
+import { PostCoordinatorData } from '../models/project';
+import { permitNoSamplingPostBody, permitNoSamplingResponseBody } from '../openapi/schemas/permit-no-sampling';
+import { postPermitNoSamplingSQL } from '../queries/permit-no-sampling-queries';
+import { getLogger } from '../utils/logger';
+import { logRequest } from '../utils/path-utils';
+
+const defaultLog = getLogger('paths/permit-no-sampling');
+
+export const POST: Operation = [logRequest('paths/permit-no-sampling', 'POST'), createNoSamplePermits()];
+
+POST.apiDoc = {
+  description: 'Creates new no sample permit records.',
+  tags: ['no-sample-permit'],
+  security: [
+    {
+      Bearer: WRITE_ROLES
+    }
+  ],
+  requestBody: {
+    description: 'No sample permits post request object.',
+    content: {
+      'application/json': {
+        schema: {
+          ...(permitNoSamplingPostBody as object)
+        }
+      }
+    }
+  },
+  responses: {
+    200: {
+      description: 'No sample permits response object.',
+      content: {
+        'application/json': {
+          schema: {
+            ...(permitNoSamplingResponseBody as object)
+          }
+        }
+      }
+    },
+    400: {
+      $ref: '#/components/responses/400'
+    },
+    401: {
+      $ref: '#/components/responses/401'
+    },
+    403: {
+      $ref: '#/components/responses/401'
+    },
+    500: {
+      $ref: '#/components/responses/500'
+    },
+    503: {
+      $ref: '#/components/responses/503'
+    },
+    default: {
+      $ref: '#/components/responses/default'
+    }
+  }
+};
+
+/**
+ * Creates new no sample permit records.
+ *
+ * @returns {RequestHandler}
+ */
+function createNoSamplePermits(): RequestHandler {
```

Comment:
question:
is it possible to add more tests to this file to address these coverage complaints?

Reply:
Definitely. Was playing with it this morning, and I might do them in a separate PR.  We don't have any testing setup for the endpoints (and handling all of the req/res/db connection stuff) so it will take a bit of trial and error to get it all setup and happy.

Reply:
ok yea let's do it later